modulus/datapipes/climate/climate_hdf5.py

```diff
@@ -0,0 +1,590 @@
+      `.npy` files with the shape (1, num_channels, 1, 1).
+
+    This pipeline assumes the HDF5 files have no metadata, such as timestamps.
+    Because of this, it's important to specify the `dt` parameter and the
+    `start_year` parameter so that the pipeline can compute the correct
+    timestamps for each timestep. These timestamps are then used to compute the
+    cosine of the solar zenith angle, if specified.
+
+    Parameters
+    ----------
+    data_dir : str
+        Directory where climate data is stored
+    stats_dir : Union[str, None], optional
+        Directory to data statistic numpy files for normalization, if None, no normalization
+        will be used, by default None
+    channels : Union[List[int], None], optional
+        Defines which climate variables to load, if None will use all in HDF5 file, by default None
+    batch_size : int, optional
+        Batch size, by default 1
+    stride : int, optional
+        Number of steps between input and output variables. For example, if the dataset
+        contains data at every 6 hours, a stride 1 = 6 hour delta t and
+        stride 2 = 12 hours delta t, by default 1
+    dt : float, optional
+        Time in hours between each timestep in the dataset, by default 6 hr
+    start_year : int, optional
+        Start year of dataset, by default 1980
+    num_steps : int, optional
+        Number of timesteps to return, by default 2 (1 for input, 1 for output)
+    lsm_filename : str, optional
+        Path to land sea mask file, by default None
+    geopotential_filename : str, optional
+        Path to geopotential file, by default None
+    use_latlon : bool, optional
+        Include latitude and longitude meshgrid, by default False
+    use_cos_zenith : bool, optional
+        Include cosine of zenith angle, by default False. If True then latitude and longitude
+        will also be computed.
+    patch_size : Union[Tuple[int, int], int, None], optional
+        If specified, crops input and output variables so image dimensions are
+        divisible by patch_size, by default None
+    num_samples_per_year : int, optional
+        Number of samples randomly taken from each year. If None, all will be use, by default None
+    shuffle : bool, optional
+        Shuffle dataset, by default True
+    num_workers : int, optional
+        Number of workers, by default 1
+    device: Union[str, torch.device], optional
+        Device for DALI pipeline to run on, by default cuda
+    process_rank : int, optional
+        Rank ID of local process, by default 0
+    world_size : int, optional
+        Number of training processes, by default 1
+    """
+
+    def __init__(
+        self,
+        data_dir: str,
+        stats_dir: Union[str, None] = None,
+        channels: Union[List[int], None] = None,
+        batch_size: int = 1,
+        stride: int = 1,
+        dt: float = 6.0,
+        start_year: int = 1980,
+        num_steps: int = 2,
```

Comment:
This might create some confusion. I suggest we have the `num_steps` to represent the number of prediction steps, and add one to that for the input.

Reply:
So I am actually much more in favor of giving a sequence of states and having the user decide what they want to do for the input output and sequence unroll. Some of the networks such as DLWP do some strange things where it takes in `n` time steps and then predicts `m` time steps. This can get pretty confusing in the loop unroll as to predict the next step you might need to take in some of the previous input time steps and some or all of the predicted time steps. I think writing a loop unroll for these cases can get very confusing when you explicitly pass the network invar and outvar.